DESIGN.md

```diff
@@ -0,0 +1,36 @@
+# HTTP Provider Design
+
+The HTTP Provider offers functionality for interacting with generic HTTP servers as part of terraform configuration.
+Specifically, the provider issues an HTTP GET request to the defined URL on every Terraform run.
+
+Below we have a collection of _Goals_ and _Patterns_: they represent the guiding principles applied during the
+development of this provider. Some are in place, others are ongoing processes, others are still just inspirational.
+
+## Goals
+
+* [_Stability over features_](.github/CONTRIBUTING.md)
+* Provide managed data source to issue HTTP GET request.
+* Support usage of requests to URLs directed to either `http` or `https`.
```

Comment:
What implementation details of HTTPS are supported? TLS 1.0 or 1.1 (😬 )? 1.3? Are there relevant IETF/W3  RFCs?

Reply:
Likely this can be also done by looking at the godoc for the HTTP client we use. And probably they link to these information there.

Reply:
Have updated `DESIGN.md`.